servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/AsyncContextMapToContextMapAdapter.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.api;
+
+import io.servicetalk.concurrent.internal.ContextMapUtils;
+import io.servicetalk.context.api.ContextMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.BiPredicate;
+import javax.annotation.Nullable;
+
+import static java.util.Objects.requireNonNull;
+
+@Deprecated
+final class AsyncContextMapToContextMapAdapter implements AsyncContextMap {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AsyncContextMapToContextMapAdapter.class);
+
+    private static final ConcurrentMap<AsyncContextMap.Key<?>, ContextMap.Key<?>> acmToCm = new ConcurrentHashMap<>();
+    private static final ConcurrentMap<ContextMap.Key<?>, AsyncContextMap.Key<?>> cmToAcm = new ConcurrentHashMap<>();
+
+    private final ContextMap contextMap;
+
+    AsyncContextMapToContextMapAdapter(final ContextMap contextMap) {
+        this.contextMap = requireNonNull(contextMap);
+    }
+
+    static <T> void newKeyMapping(final AsyncContextMap.Key<T> acmKey, final ContextMap.Key<T> cmKey) {
+        final ContextMap.Key<?> oldCmKey = acmToCm.putIfAbsent(acmKey, cmKey);
+        final AsyncContextMap.Key<?> oldAcmKey = cmToAcm.putIfAbsent(cmKey, acmKey);
+        if (oldCmKey != null || oldAcmKey != null) {
+            LOGGER.warn("Tried to register a new mapping of AsyncContextMap.Key={} to {}, but other mappings already " +
```

Comment:
"AsyncContextMap.Key={} to {}" -> "AsyncContextMap.Key={} to ContextMap.Key={}".
Same below?

Reply:
No, the `ContextMap#toString()` implementations already print it consistently using a shared utility: 
https://github.com/apple/servicetalk/blob/26422543e8ce5e4b03aa65f610641ba5a5e46d7a/servicetalk-concurrent-internal/src/main/java/io/servicetalk/concurrent/internal/ContextMapUtils.java#L41-L42